drivers/entropy/entropy_mcxw.c

```diff
@@ -0,0 +1,63 @@
+/* entropy_mcxw.c - NXP MCXW entropy driver */
+
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_mcxw_trng
+
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/entropy.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/util.h>
+#include <soc.h>
+
+#include "sss_crypto.h"
+
+struct entropy_mcxw_data_str {
+	struct k_sem sem_lock;
+};
+
+static struct entropy_mcxw_data_str entropy_mcxw_data;
+
+static int entropy_mcxw_get_entropy(const struct device *dev, uint8_t *buf, uint16_t len)
+{
+	sss_sscp_rng_t ctx;
+	int result = -EIO;
+
+	__ASSERT_NO_MSG(buf != NULL);
+	__ASSERT_NO_MSG(&entropy_mcxw_data == dev->data);
+
+	k_sem_take(&entropy_mcxw_data.sem_lock, K_FOREVER);
+
+	if (CRYPTO_InitHardware() != kStatus_Success) {
+		goto exit;
+	}
+	if (sss_sscp_rng_context_init(&g_sssSession, &ctx, 0u) != kStatus_SSS_Success) {
+	} else if (sss_sscp_rng_get_random(&ctx, buf, len) != kStatus_SSS_Success) {
+	} else if (sss_sscp_rng_free(&ctx) != kStatus_SSS_Success) {
```

Comment:
What is the point of these empty checks, you should be able to create a proper `if` check here?